druntime/src/core/sys/linux/linux/if_packet.d

```diff
@@ -0,0 +1,318 @@
+    PACKET_DROP_MEMBERSHIP = 2,
+    PACKET_RECV_OUTPUT     = 3,
+
+    PACKET_RX_RING         = 5,
+    PACKET_STATISTICS      = 6,
+    PACKET_COPY_THRESH     = 7,
+    PACKET_AUXDATA         = 8,
+    PACKET_ORIGDEV         = 9,
+    PACKET_VERSION         = 10,
+    PACKET_HDRLEN          = 11,
+    PACKET_RESERVE         = 12,
+    PACKET_TX_RING         = 13,
+    PACKET_LOSS            = 14,
+    PACKET_VNET_HDR        = 15,
+    PACKET_TX_TIMESTAMP    = 16,
+    PACKET_TIMESTAMP       = 17,
+    PACKET_FANOUT          = 18,
+    PACKET_TX_HAS_OFF      = 19,
+    PACKET_QDISC_BYPASS    = 20,
+    PACKET_ROLLOVER_STATS  = 21,
+    PACKET_FANOUT_DATA     = 22,
+    PACKET_IGNORE_OUTGOING = 23,
+    PACKET_VNET_HDR_SZ     = 24,
+
+    PACKET_FANOUT_HASH     = 0,
+    PACKET_FANOUT_LB       = 1,
+    PACKET_FANOUT_CPU      = 2,
+    PACKET_FANOUT_ROLLOVER = 3,
+    PACKET_FANOUT_RND      = 4,
+    PACKET_FANOUT_QM       = 5,
+    PACKET_FANOUT_CBPF     = 6,
+    PACKET_FANOUT_EBPF     = 7,
+
+    PACKET_FANOUT_FLAG_ROLLOVER        = 0x1000,
+    PACKET_FANOUT_FLAG_UNIQUEID        = 0x2000,
+    PACKET_FANOUT_FLAG_IGNORE_OUTGOING = 0x4000,
+    PACKET_FANOUT_FLAG_DEFRAG          = 0x8000,
+}
+
+struct tpacket_stats
+{
+    uint tp_packets;
+    uint tp_drops;
+}
+
+struct tpacket_stats_v3
+{
+    uint tp_packets;
+    uint tp_drops;
+    uint tp_freeze_q_cnt;
+}
+
+struct tpacket_rollover_stats
+{
+    align(8):
+    ulong tp_all;
+    ulong tp_huge;
+    ulong tp_failed;
+}
+
+union tpacket_stats_u
+{
+    tpacket_stats    stats1;
+    tpacket_stats_v3 stats3;
+}
+
+struct tpacket_auxdata
+{
+    uint   tp_status;
+    uint   tp_len;
+    uint   tp_snaplen;
+    ushort tp_mac;
+    ushort tp_net;
+    ushort tp_vlan_tci;
+    ushort tp_vlan_tpid;
+}
+
+enum : uint
+{
+    TP_STATUS_KERNEL          = 0,
+    TP_STATUS_USER            = 1 << 0,
+    TP_STATUS_COPY            = 1 << 1,
+    TP_STATUS_LOSING          = 1 << 2,
+    TP_STATUS_CSUMNOTREADY    = 1 << 3,
+    TP_STATUS_VLAN_VALID      = 1 << 4,
+    TP_STATUS_BLK_TMO         = 1 << 5,
+    TP_STATUS_VLAN_TPID_VALID = 1 << 6,
+    TP_STATUS_CSUM_VALID      = 1 << 7,
+    TP_STATUS_GSO_TCP         = 1 << 8,
+}
+
+enum : uint
+{
+    TP_STATUS_AVAILABLE    = 0,
+    TP_STATUS_SEND_REQUEST = 1 << 0,
+    TP_STATUS_SENDING      = 1 << 1,
+    TP_STATUS_WRONG_FORMAT = 1 << 2,
+}
+
+enum : uint
+{
+    TP_STATUS_TS_SOFTWARE     = 1 << 29,
+    TP_STATUS_TS_RAW_HARDWARE = 1U << 31,
+}
+
+enum uint TP_FT_REQ_FILL_RXHASH = 0x1;
+
+struct tpacket_hdr
+{
+    ulong   tp_status;
+    uint    tp_len;
+    uint    tp_snaplen;
+    ushort  tp_mac;
+    ushort  tp_net;
+    uint    tp_sec;
+    uint    tp_usec;
+}
+
+enum TPACKET_ALIGNMENT = 16;
+size_t TPACKET_ALIGN(size_t x) { return (x + TPACKET_ALIGNMENT - 1) &~ (TPACKET_ALIGNMENT - 1); }
+enum TPACKET_HDRLEN = TPACKET_ALIGN(tpacket_hdr.sizeof) + sockaddr_ll.sizeof;
+
+struct tpacket2_hdr
+{
+    uint     tp_status;
```

Comment:
This is unsigned long in the original header, which is I believe 64 bit on 64-bit arch?

Reply:
It looks like I did screw up the one above though, since it's `ulong` instead of `c_ulong`.

Reply:
Fixed